torch/csrc/jit/serialization/export.cpp

```diff
@@ -333,21 +341,19 @@ EncoderBase::EncoderBase(
   model_proto_.set_producer_version(onnx_torch::PRODUCER_VERSION);
 }
 
-void EncoderBase::EncodeValueInfo(
-    onnx::GraphProto* graph_proto,
-    onnx::ValueInfoProto* v,
+void EncoderBase::EncodeValueInfoType(
```

Comment:
Is this part only refactoring? or is there some logic changes?

Reply:
To extend EncodeValueInfo to support more complex (nested) types, I refactored the part of the code that encodes type as a recursive function.